app/services/notification/ScheduledNotificationRunner.scala

```diff
@@ -0,0 +1,40 @@
+package services.notification
+
+import config.AMIableConfig
+import models.{Instance, Owner, SSA}
+import play.api.Logger
+import prism.{Prism, PrismLogic}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+object ScheduledNotificationRunner {
+
+  def run(mailClient: AWSMailClient)(implicit config: AMIableConfig, ec: ExecutionContext): Future[Any] = {
```

Comment:
That said, if you use `Attempt` throughout the program instead of mixing it with Futures you can write this really neatly with a for-comp. I had a fiddle and came up with this:

```scala
  def run(mailClient: AWSMailClient)(implicit config: AMIableConfig, ec: ExecutionContext): Attempt[List[String]] = {
    for {
      instancesWithAmis <- Prism.instancesWithAmis(SSA(stage = Some("PROD")))
      oldInstances = PrismLogic.oldInstances(instancesWithAmis)
      _ = Logger.debug(s"Found ${oldInstances.size} PROD instances running on an out-of-date AMI")
      owners <- Prism.getOwners
      mailIds <- Attempt.traverse(owners)(owner => mailClient.send(owner, instancesForOwner(owner, oldInstances)))
    } yield mailIds
  }
```

This required changing `AWSMailClient.send` to return an `Attempt[String]` instead of a `Future[String]` like it does now.

Most of the error logs duplicate logs that are written elsewhere in the program so they probably aren't needed. I've left the debug one in the example!